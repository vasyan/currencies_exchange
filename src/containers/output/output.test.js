import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import configureMockStore from 'redux-mock-store'
import Output from './index'

jest.mock('utils/getFieldStyles', () =>
  jest.fn().mockImplementation(() => ({ fontSize: 42 }))
)

describe('<Output>', () => {
  const store = configureMockStore()({
    widgets: {
      exchange: {
        currencyFrom: 'FOO',
        currencyTo: 'BAR',
        amount: '42'
      }
    },
    collections: {
      rates: {
        USD: {
          FOO: 0.4242,
          BAR: 0.2424
        }
      }
    }
  })
  it('should render', () => {
    expect(
      shallowToJson(shallow(<Output store={store} />).dive())
    ).toMatchSnapshot()
  })
})
