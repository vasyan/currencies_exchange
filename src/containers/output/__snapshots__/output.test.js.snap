// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Output> should render 1`] = `
<div>
  <CurrencySelect
    currencyName="BAR"
    onNext={[Function]}
    onPrev={[Function]}
  >
    <div
      className="value_wrapper"
    >
      <span
        className="value"
        style={
          Object {
            "fontSize": 42,
          }
        }
      >
        23.9999
      </span>
    </div>
  </CurrencySelect>
  <div
    className="rate_wrapper"
  >
    <span>
      1 
      FOO
       = 
      0.5714
       
      BAR
    </span>
  </div>
</div>
`;
